$OpenBSD: patch-rake_d_helpers_rb,v 1.1 2015/05/27 21:18:48 ajacoutot Exp $

build system: force UTF-8 string encoding when reading source files

--- rake.d/helpers.rb.orig	Tue May 26 14:14:05 2015
+++ rake.d/helpers.rb	Tue May 26 14:14:37 2015
@@ -237,7 +237,7 @@ def remove_files_by_patters patterns
 end
 
 def read_files *file_names
-  Hash[ *file_names.flatten.collect { |file_name| [ file_name, IO.readlines(file_name) ] }.flatten(1) ]
+  Hash[ *file_names.flatten.collect { |file_name| [ file_name, IO.readlines(file_name).collect { |line| line.force_encoding "UTF-8" } ] }.flatten(1) ]
 end
 
 class Rake::Task
